### DISCUSSION TITLE
Verifier of Query Results

### DISCUSSION THREAD

**Discussion by dzobbe** [spaceandtimefdn/sxt-proof-of-sql#79] [Q&A] [answered]
Hi all,

I have seen the example generating the proof and doing the verification. I need some help in building a verifier only code that uses the output received by SxT, e.g.: 
```
{"requestId":"a88861df-64d2-44b6-8d6c-a5fe61f5a316","requestTimestamp":"2024-08-02T17:35:55.120001497Z","metadata":{"verificationHash":"xJU5uaAUFgxuoZJCpFged0AbxI7N6UWhvs6hAqer6ks="},"data":[{"att_data":"Sample attestation data","proof_order":1234567890}]} 
```
How can I use that verification hash and the data to verify them?

Thanks in advance.

**Comment by JayWhite2357** [comment]
Hey @dzobbe,
Great question. At the moment the SxT endpoint that you are getting that response from does not return that proof, but instead does verification as a service. We tried to make the API endpoint similar to the non-tamperproof pathway.

However, I'm happy to add the ability to return the proof from that endpoint as well, which will enable what you want. I'll post an update when we have it done.

**Reply by dzobbe** [reply]
Ah thanks a lot. So essentially this `/v2/sql/tamperproof-query` is already returning the result of the verification. I didn't get it. 
Yeah, I think it would be useful to receive just the proof commitment. I would also suggest to create a sample code that does proof generation and proof verification in two separate programs. That can simplify the life of sxt users ;)
Thanks @JayWhite2357

**Reply by dzobbe** [reply]
hey @JayWhite2357 any update on this?

**Reply by JayWhite2357** [reply]
@dzobbe This is almost done. We should have a new endpoint either this Monday or next. I will post back here when have it available.

**Reply by dzobbe** [reply]
hey man. Nothing?

**Reply by JayWhite2357** [reply]
Here you go:
https://github.com/spaceandtimelabs/sxt-proof-of-sql-sdk
It's still slightly in progress, but should be stabilizing quickly now that all the pieces are in place.

**Reply by dzobbe** [reply]
I think it is a private repo since I can't see it

**Reply by JayWhite2357** [reply]
It should be public now

**Comment by JayWhite2357** [comment, marked as answer]
Adding an update here: we've added a couple more examples in the SDK repo on how to use the SXTChain. Take a look here: https://github.com/spaceandtimelabs/sxt-proof-of-sql-sdk/tree/main/examples/deno for one example using javascript.
